### PR TITLE
API for pre-generating chunks inside a world border

### PR DESCRIPTION
**Pull request by DDoS**
For #882, implemented in https://github.com/SpongePowered/SpongeCommon/pull/328

@bloodmc

### CODE REVIEW

File: src/main/java/org/spongepowered/api/world/WorldBorder.java [hunk 1]
```diff
@@ -194,4 +201,109 @@
...
+         *
+         * <p>Optional.</p>
+         *
+         * @param logger A logger for the generator
+         * @return This for chained calls
+         */
+        ChunkPreGenerate logger(@Nullable Logger logger);
+
+        /**
+         * <p>Sets the interval between generation runs.</p>
```
Comment (ST-DDT) [inline src/main/java/org/spongepowered/api/world/WorldBorder.java]: IMO the first sentence does not need to encapsulated in `<p>`.

File: src/main/java/org/spongepowered/api/world/WorldBorder.java [hunk 2]
```diff
@@ -194,4 +201,109 @@
      * @param damage The damage amount
      */
     void setDamageAmount(double damage);
+
+    /**
+     * Returns a new builder for creating a task to pre-generate the chunks
+     * inside the border. The current state of the border is "snapshoted" for
+     * the builder. It is not backed by this instance. If the border size is
+     * changing then the target size is used.
```
Comment (ST-DDT) [inline src/main/java/org/spongepowered/api/world/WorldBorder.java]: the new or the old target size?

Comment (DDoS) [inline src/main/java/org/spongepowered/api/world/WorldBorder.java]: The target is by definition the new size. The "target" as in "the size that is going to be reached".

**Comment by DDoS** [comment]
Can I merge this and the implementation tonight?

**Pull request closed without merging**